### DISCUSSION TITLE
How to use less in NextJs 13.x?

### DISCUSSION THREAD

**Discussion by hqwlkj** [vercel/next.js#45169] [Help]
### Describe the feature you'd like to request

Hope `NextJs` doesn't restrict users to only use the built-in CSS support.

### Describe the solution you'd like

It can be extended by means of plug-ins.

### Describe alternatives you've considered

The currently used `next-with-less` does not support NextJs 13.x.

**Comment by SouthLink** [comment]
I am using this plugin for my project and feel good, next version 13.1.1

**Comment by hqwlkj** [comment]
> I am using this plugin for my project and feel good, next version 13.1.1

May I ask which plugin you are using?

**Comment by SouthLink** [comment]
next-with-less

> > 我正在为我的项目使用这个插件并且感觉很好，下一个版本 13.1.1
> 
> 请问你用的是什么插件？

next-with-less

**Reply by a8655099449** [reply]
`next-with-less`  Cannot be used in next13 at all

**Reply by SouthLink** [reply]
> `next-with-less` Cannot be used in next13 at all

this is my next.config.js , It feels good
// next.config.js
```
/** @type {import('next').NextConfig} */
const withLess = require("next-with-less");
const path = require('node:path');

const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  experimental: {
    appDir: true,
  },
  transpilePackages: ['antd-mobile'],
  webpack(config, options) {
    // disable css-module in Next.js
    config.module.rules.forEach((rule) => {
      const { oneOf } = rule;
      if (oneOf) {
        oneOf.forEach((one) => {
          if (!`${one.issuer?.and}`.includes('_app')) return;
          one.issuer.and = [path.resolve(__dirname)];
        });
      }
    })

    return config;
  }
}

module.exports = withLess(nextConfig);
```

**Reply by Jairwin-L** [reply]
It seems that the latest version（13.2.1） does not support this plugin？

**Reply by SouthLink** [reply]
My version is 13.1.1. What is wrong with your version? Could you please send it

**Reply by Jairwin-L** [reply]
my version: 13.2.1
example：
page.tsx
<img width="411" alt="image" src="https://user-images.githubusercontent.com/32570960/222162139-ca33af03-b8ab-472c-a273-2be909fa10e7.png">
page.less
<img width="249" alt="image" src="https://user-images.githubusercontent.com/32570960/222162284-b13ff733-4580-4140-9930-c213672e2fed.png">
the page is finally displayed, expecting red, but actually without any color
<img width="699" alt="image" src="https://user-images.githubusercontent.com/32570960/222162650-d248ca25-f3d6-4e31-a2f2-525f2c38f636.png">
my config
<img width="453" alt="image" src="https://user-images.githubusercontent.com/32570960/222163913-f04d728f-dcf5-4c46-bd85-356e42458c1f.png">

**Reply by SouthLink** [reply]
APP directory is the default server component, why not support the introduction of global css I think it should be related to this RFC: https://github.com/vercel/next.js/discussions/27953#discussioncomment-3978605, you should try with CSS module mode to the introduction of CSS, or put the file on pages directory

**Comment by Jairwin-L** [comment]
@ijjk  hope that the official can continue to support less, although this plugin is no longer supported because of low usage.
https://github.com/vercel/next-plugins/tree/master/packages/next-less

**Comment by sunkid1995** [comment]
+ 1 i have the same problem

**Comment by HUANGXUANKUN** [comment]
we need less support badly, as some of external packages rely on that

**Reply by zhunh** [reply]
have you got a solution for this, cause i have the same situation

**Comment by HaimAbeles** [comment]
any solution?

**Reply by hqwlkj** [reply]
You can use or refer to this example： [next-antd](https://github.com/hqwlkj/next-antd)

**Reply by HaimAbeles** [reply]
I need an example of how to use theme.config, it is not here in the project

**Comment by MouhtaramSoufiane** [comment]
i have a problem i want to add support less to next configuration (next v14)

when i add less-loader config i get css is diabled

how can i fix this issue plaise guys.
